File: src/test/testdata/inspections/unsafeIterators/IntMap.java
package main;

import com.badlogic.gdx.utils.IntIntMap;
import com.badlogic.gdx.utils.IntMap;

class IntMapTest {

  private IntMap<String> map = new IntMap();

  void init() {
    map.put(1, "One");
    map.put(2, "Two");
    map.put(3, "Three");
  }

  void print() {
    for (IntMap.Entry e: <warning>map</warning>) {
      System.out.println(e.value);
    }
  }

  void print2() {
    for (IntMap.Entry<String> e: new IntMap.Entries<String>(map)) {
      System.out.println(e.value);
    }
  }

  void test() {
    int[] i = <warning>map.keys()</warning>.toArray().toArray();
    IntMap map2 = map;
    Object o = map2;
    <warning>((IntMap)o).entries()</warning>;
    <warning>((IntMap)o).iterator()</warning>;
    <warning>((IntMap)o).values()</warning>;
  }

}

File: src/test/testdata/inspections/unsafeIterators/IntMap.kt
package main

import com.badlogic.gdx.utils.IntIntMap
import com.badlogic.gdx.utils.IntMap

@Suppress("UNUSED_VARIABLE", "UNCHECKED_CAST", "USELESS_CAST")
internal class IntMapTestKotlin {

  private val map = IntMap<String>()

  fun init() {
    map.put(1, "One")
    map.put(2, "Two")
    map.put(3, "Three")
  }

  fun print() {
    for (e in <warning>map</warning>) {
      println(e.value)
    }
  }

  fun print2() {
    for (e in IntMap.Entries<String>(map)) {
      println(e.value)
    }
  }

  fun test() {
    val i = <warning>map.keys()</warning>.toArray().toArray()
    val map2 = map
    val o: Any = map2
    <warning>(o as IntMap<String>).entries()</warning>
    <warning>(o as IntMap<String>).iterator()</warning>
    <warning>(o as IntMap<String>).values()</warning>

    val map3 = if (true==false) map2 else null

    map3?.let {
      <warning>it.keys()</warning>
      it.let { bla ->
        <warning>bla.values()</warning>
      }
    }
  }

}
